lib/src/main/scala/spinal/lib/bus/avalon/AvalonST.scala

```diff
@@ -0,0 +1,192 @@
+package spinal.lib.bus.avalon
+
+import spinal.core._
+import spinal.lib._
+
+case class AvalonSTConfig(dataWidth: Int,
+                          useData: Boolean = true,
+                          useChannels: Boolean = false,
+                          channelWidth: Int = -1,
+                          useError: Boolean = false,
+                          errorWidth: Int = -1,
+                          useReady: Boolean = true,
+                          useValid: Boolean = true,
+                          useEmpty: Boolean = false,
+                          emptyWidth: Int = -1,
+                          useEOP: Boolean = false,
+                          useSOP: Boolean = false,
+                          /* Meta data values */
+                          maxChannels: Int = -1,
+                          beatsPerCycle: Int = -1,
+                          dataBitsPerSymbol: Int = -1,
+                          emptyWithinPacket: Boolean = false,
+                          errorDescriptor: Array[String] = Array(),
+                          symbolEndianness: Endianness = LITTLE,
+                          readyLatency: Int = 0,
+                          readyAllowance: Int = 0) {
+  assert(!(useReady || useValid), "AvalonST must have at least ready OR valid enabled!")
+}
+
+case class AvalonSTPayload(config: AvalonSTConfig) extends Bundle {
+  val data: Bits    = if (config.useData) Bits(config.dataWidth bit) else null
+  val channel: UInt = if (config.useChannels) UInt(config.channelWidth bit) else null
+  val error: Bits   = if (config.useError) Bits(config.errorWidth bit) else null
+  val empty: UInt   = if (config.useEmpty) UInt(config.emptyWidth bit) else null
+  val eop: Bool     = if (config.useEOP) Bool() else null
+  val sop: Bool     = if (config.useSOP) Bool() else null
+}
+
+case class AvalonST(config: AvalonSTConfig) extends Bundle with IMasterSlave {
+
+  val ready: Bool = if (config.useReady) Bool() else null
+  val valid: Bool = if (config.useValid) Bool() else null
+  val payload: AvalonSTPayload = AvalonSTPayload(config)
+
+  lazy val latencyDelay: Bool = Delay(this.ready, config.readyLatency)
```

Comment:
Hmm lazy val in a bundle, i'm realy not sure SpinalHDL will like it.
The goal of those signals is just to cache some values if used right ? Not realy to carry them along the bundle right ?

Reply:
Missed this. Yeah those are internal for ready tracking. Avalon as some weird timing relationships. I can un-lazy them if you think there might be problems.
Completely forgot up to now that I was working in a Bundle. XP

Reply:
Hi ^^
There is still some lazy val, can you un-lazy them ?
If you want, there is the "signalCache" API.

Reply:
Finally done. Let me know if I'm using signalCache right.